docs/esm-views/how-to-build.md

```diff
@@ -0,0 +1,44 @@
+---
+parent: ESM Views
+nav_order: 1
+title: How to build ESM Views
+---
+
+# How to build ESM Views
+
+ESM views are built with the [`modular build`](../commands/build.md) command.
+The default behaviour when building an ESM view is:
+
+1. All the non-local `import`s in the package's `src` directory are extracted
+   and matched with their version in the package's `package.json` or the
+   `package.json` in the repository root and with their exact version in the
+   repo's lockfile.
+2. The main entrypoint (as defined in the ESM view's `package.json`'s `main`
+   field) and its local imports are bundled in a single file.
+3. All the `import` statements to non-local dependencies encountered in the
+   process are rewritten to `import` from an external ESM CDN (by default
+   https://www.skypack.dev/), using the versions extracted in step 1. By
+   default, versions extracted from `package.json` will be used, but users can
+   customize the rewrite template to use versions from the lockfile instead.
+4. All the local CSS is bundled in a single file.
+5. The `dist` directory is generated, containing:
+   - The js file
+   - The css file
+   - A [package manifest](./output-package-manifest.md) (`package.json`) file
+     containing:
+     - The location of the js bundle (`"module"` field)
+     - The location of the css bundle (`"style"` field)
+     - An object with the whole set of dependencies and their version ranges
+       (`"dependencies"` field)
+     - An array of bundled dependencies (`bundledDependencies` field)
+   - A synthetically generated `index.html` file, linking the trampoline file
+     and the css bundle
+   - A synthetically generated trampoline file, dynamically `import`ing the js
```

Comment:
Nit: on a previous team, I called this sandbox. A bit like what codesandbox or codepen give you - a shell around your app.

I think having a specific name (trampoline) is fine, but maybe this could be a bit more descriptive or explicit? From the docs as you have it, it's not entirely clear if this trampoline forms part of your app / shipped assets or not.

Reply:
Good point. Updated to specify that `_trampoline.js` is a file and in which locations of the `dist` directory the trampoline and the index are generated.